plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/NonSystemJobSuspender.java

```diff
@@ -45,41 +45,48 @@ private SuspendedJob(Job job, long scheduleDelay) {
     }
   }
 
+  private static final AtomicBoolean suspended = new AtomicBoolean(false);
+
   private static final List<SuspendedJob> suspendedJobs = new ArrayList<>();
 
   private static final NonSystemJobScheduleListener jobScheduleListener =
       new NonSystemJobScheduleListener();
 
   /** Once called, it is imperative to call {@link resume()} later. */
-  public static synchronized void suspendFutureJobs() {
-    Preconditions.checkState(!suspended, "Already suspended.");
-    suspended = true;
+  public static void suspendFutureJobs() {
```

Comment:
The use of an `AtomicBoolean` ensures that there can only be a single winner.
If `suspended` was previously `true` then `Preconditions` will throw an exception.

If we were going to do anything, I would change `suspended` to be a ref-counter so that `suspendFutureJobs()` and `resume()` can be called multiple times.